### ISSUE TITLE
GIOU, DIOU,CIOU 

### ISSUE THREAD

**Issue by hejinsome** [facebookresearch/detectron2#1085]
Hello! detectron2 use IOU loss, that is, smooth_l1_loss function.  will you plan to support of GIOU,DIOU,CIOU in detectron2?

**Comment by WeihongM** [comment]
any updates?

**Comment by VadymBoikov** [comment]
There is a detectron2 fork, which implemented GIOU https://github.com/generalized-iou/Detectron.pytorch

**Comment by ferna11i** [comment]
Hi!

I've implemented DIoU and CIoU in my clone of Detectron2. However, I'm not sure if I have got all the mathematical details correct. That being said, I did get improvements on certain private datasets over L1 loss. Here's my repo: https://github.com/ferna11i/detectron2

Looking for anyone who would like to work with me to solve this issue. We could potentially add these new loss functions to Detectron2.

The major changes have been done on two files:
1. detectron2/modeling/roi_heads/fast_rcnn.py
2. detectron2/config/defaults.py

Right off the bat I can point out definite changes that would need to be done such as:
1. Confirm the mathematics behind the functions (that I've worked on, not the original authors code, I'm certain their code is right) is working as expected.
2. Better way to reference the config details in fast_rcnn.py


Looking forward to working with anyone who is interested in adding IOU based loss functions to Detectron2.

**Comment by jonmorton** [comment]
Hey @hejinsome , @WeihongM , @ferna11i - 

Support for GIoU loss has been added in 73bbd7c1828d5d7560b380c69379fcaa0b226dda for the rcnn head box regression loss and in af0462a8632e4391c70ce45eb06080cb43e29206 for the rpn box regression loss. 

See https://github.com/facebookresearch/detectron2/blob/master/configs/COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_1x_giou.yaml for a sample of enabling GIoU.

**Comment by ferna11i** [comment]
Thank you @jonmorton. This is a good feature to have in the detectron2 library. 

I just have one question in terms of in the implementation. The GIOU loss here uses Bounding Boxes while the original authors code used delta values. Have you noticed any difference in performance of using boxes instead of delta values?

**Comment by cbasavaraj** [comment]
Hi @jonmorton I noticed you've set`BBOX_REG_LOSS_WEIGHT: 2.0` for RPN and 10.0 for ROI. Is this just to show we can alter them, or any good reason for these values? Would you recommend staying with the defaults of 1.0 even when training with GIOU? Thanks

**Comment by petoor** [comment]
The number 10 is from https://arxiv.org/abs/1902.09630 section 4.2. However i cant seem to find reg loss to be 2 anywhere.  In https://github.com/generalized-iou/Detectron.pytorch they use the value 1.

If the DIOU loss seems to be working would you mind making a pull request? I am very interested in this feature as well.
